broker/src/main/java/io/zeebe/broker/system/partitions/impl/steps/ExporterDirectorPartitionStep.java

```diff
@@ -34,7 +34,14 @@
 
     final ExporterDirector director = new ExporterDirector(exporterCtx);
     context.setExporterDirector(director);
-    return director.startAsync(context.getScheduler());
+    final var started = director.startAsync(context.getScheduler());
+    started.onComplete(
+        (nothing, error) -> {
+          if (error == null && !context.shouldExport()) {
+            director.pauseExporting();
```

Comment:
Is this maybe problematic? That we might have an race condition where we can export maybe one record because we are so fast?

Reply:
Might make sense to give the state as parameter of the startAsync?

Reply:
That's right. I pass it as an argument to the constructor now. This will be a problem in StreamProcessor as well, because it also follows the same pattern.

Reply:
Do you want to create a follow up issue for that? Or do you want to do it in this pr?

Reply:
I will create a follow up issue.